Uninett/Argus-frontend
# Fix feide logout

Fixes #340.

**Cause of the bug:**
Login via dataporten  sent responses with a "Set-Cookie" header. Backend forwards it and frontend sets cookie sent from dataporten. Frontend sets cookies from dataporten as a subdomain cookie and will not remove it, unless that specific subdomain is provided on removal request. Subdomain format is ".COOKIE_DOMAIN". Presence of "token" in cookies is equal to user being considered logged on the backend, and is a main factor in deciding on whether to call either token-auth, or auth endpoint. 

**Introduces changes:**

- requires a new env variable (REACT_APP_COOKIE_DOMAIN) in frontend config. This one is **necessary** for removal of token which is set during dataporten-auth, and is a nice-to-have for other auth methods. This is due to the fact that removal of token set during dataporten auth, domain has to be provided. Domain could be extracted from the url, but is not optimal. Cookie domain can be set in the api settings, and can theoretically be different than api url (see domain values for cookies in the review stage in argus-run).
- cookies are properly removed on logout, regardless of auth method
- post-logout cleanup is improved

**Functionality:**

- User can have several active tabs/windows of Argus. Logout in one of them is reflect in all active tabs. Restrictions apply:
      - Tabs/windows can not be of different browsers, nor same browser but incognito mode. This is due to backend not supporting several sessions of the same user. 
- Login event in a new browser, while there is an active session in another browser, will work but will also trigger logout in another browser. This does not apply if login method is different in different browsers (f.e. userpass in one, and dataporten in another).

**Manual testing suggestion:**

1. Log in via userpass/dataporten.
2. Open f.e. /incidents in another browser
3. Open f.e /incidents in a new window of the same browser
4. Log out in one of the active tabs/windows
5. Observe, one of the following (depending on auto-update method):
       - logout is triggered immediately in all other active tabs/windows
       - logout is triggered as soon as user interacts with the app in the rest of the tabs/windows (no auth-restricted operations can be performed)
6. Observe redirect to /login
7. Try go to f.e. /incidents without performing login
8. Observe immediate redirect back to /login
